### PR TITLE
Update megaparsec library to 6.0.0

### PR DESCRIPTION
**Pull request by CristhianMotoche**
I need to read a little bit about the library and the changes in the new version.

### CODE REVIEW

**Comment by mrkkrp** [comment]
Your types are wrong, I have added a note to the docs, because it looks like this is a common confusion: https://github.com/mrkkrp/megaparsec/commit/dbc7f52daaad78e81ae8ef4565b51ace356db1da

File: Text/Inflections.hs [hunk 1]
```diff
@@ -134,6 +133,9 @@ import Text.Megaparsec
 
 #if MIN_VERSION_base(4,8,0)
 import Prelude hiding (Word)
+data Void deriving (Eq, Ord, Show)
```
Comment (mrkkrp) [inline Text/Inflections.hs]: Don't do this! This `Void` is not the same as the one Haskell ecosystem uses. Add a conditional dependency on the `void` package to the Cabal file:

```
  if !impl(ghc >= 7.10)
    build-depends:    void         == 0.7.*
```

File: inflections.cabal [hunk 2]
```diff
@@ -47,7 +47,7 @@ library
     ghc-options:      -Wall -Werror
   else
     ghc-options:      -O2 -Wall
-  build-depends:       base         >= 4.6   && < 5.0
+  build-depends:       base         >= 4.7   && < 5.0
```
Comment (mrkkrp) [inline inflections.cabal]: Why?

File: Text/Inflections.hs (hunk 1 above)
Comment (CristhianMotoche) [inline Text/Inflections.hs]: Fixing it! Thanks.

**Comment by CristhianMotoche** [comment]
@mrkkrp Now that Megaparsec 6.0.0 doesn't define the instance of Arbitrary for `ParseError` should we define it in our tests with a wrapper?

**Comment by mrkkrp** [comment]
It's OK to define it in the test suite as orphan instance.

**Comment by CristhianMotoche** [comment]
@mrkkrp @sestrella This is working for `GHC >= 7.8`. What should we do with `GHC 7.6`?

**Comment by mrkkrp** [comment]
Ah, I forgot we dropped support for 7.6 in Megaparsec after version 5.3.0 (if I remember correctly). I guess you can only drop support for that version of GHC for `inflections`, no other choice.

**Comment by CristhianMotoche** [comment]
@mrkkrp @sestrella Could you take a look, please?

**Comment by sestrella** [comment]
@CristhianMotoche please rebase this branch and open a PR against master directly

File: test/Text/InflectionsSpec.hs [hunk 3]
```diff
@@ -39,8 +89,9 @@ spec = do
     context "when given a parse error" $
       it "throws the correct exception" $
         property $ \err ->
```
Comment (sestrella) [inline test/Text/InflectionsSpec.hs]: @CristhianMotoche you don't need to define all those newtypes. Use [forAll](https://hackage.haskell.org/package/QuickCheck-2.10.0.1/docs/Test-QuickCheck.html#v:forAll) instead:


```HS
arbitraryParseError ::   Gen (ParseError Char Void)
arbitraryParseError = ...


property "..." $ forAll arbitraryError $ \err -> 
  betterThrow (Left err) `shouldThrow ` ...

````

**Comment by CristhianMotoche** [comment]
Closing in favor of #51

**Pull request closed without merging**